core/src/avm2/optimizer/dce.rs

```diff
@@ -0,0 +1,51 @@
+use crate::avm2::op::Op;
+
+use std::cell::Cell;
+use std::collections::HashSet;
+
+pub fn eliminate_dead_code(ops: &[Cell<Op<'_>>], jump_targets: &HashSet<usize>) {
+    // Clippy's suggestion makes the code less readable
+    #![expect(clippy::needless_range_loop)]
+
+    // Remove jumps and the code that they jump over when the code inside the
+    // jump is unreachable
+    for (i, op) in ops.iter().enumerate() {
+        match op.get() {
+            Op::Jump { offset } => {
+                let mut is_reachable = false;
+                for position in i..offset {
+                    if jump_targets.contains(&position) {
+                        // If this position (inside the ops being jumped over) is
+                        // unreachable, we can exit
+                        is_reachable = true;
+                    }
+                }
+
+                if !is_reachable {
+                    for position in i..offset {
+                        // Set all unreachable ops to Nop, including the Jump op
+                        ops[position].set(Op::Nop);
+                    }
+                }
+            }
+            Op::PopJump { offset } => {
```

Comment:
The code duplication between the two match arms is a little unfortunate, is there no way to merge them?

Reply:
Hm, you're right, it's fine now ^^